packages/indexer-agent/src/commands/start.ts

```diff
@@ -721,6 +722,19 @@ export default {
       )
     }
 
+    // Validate if the Network Subgraph belongs to the current provider's network.
+    // This check must be performed after we ensure the Network Subgraph is being indexed.
+    try {
+      await validateNetworkId(
+        networkMeta,
+        argv.networkSubgraphDeployment,
```

Comment:
would it be nice to check `argv.networkSubgraphEndpoint` for its deployment hash if indexer isn't syncing the network subgraph? Maybe we could expect everyone to be indexing it once there's no hosted service endpoint?

Reply:
I think it would be an excellent check! 

I thought we always enforced that the Network Subgraph should be synced, but now I see that `argv.networkSubgraphDeployment` can be null.
I'll find out where we could insert that check.

Reply:
Upon reflection, testing the hash of the Network Subgraph is not feasible, as it will only be not indexed if the indexer does not provide its hash. 

Unless I'm missing something, we would not have a reference value to check even if we attempted to query the deployment hash from the endpoint.

Reply:
I think this surfaced the fact that we only run `validateNetworkId` if the indexer syncs the Network Subgraph. 
We should also validate when it does not.

Reply:
If we want to graph the valid hash of the Network Subgraph when the indexer doesn't provide deploymentHash nor sync the network subgraph but did provide an endpoint, then Network Subgraph would make a request (`query { _meta { deployment } }`) to the endpointClient. I think the decision is made [here](https://github.com/graphprotocol/indexer/blob/1da01fc5b268aa8bd54c604f6d85befc4e39c59d/packages/indexer-common/src/network-subgraph.ts#L149). Do you think this would help?

Reply:
Thanks!
This looks like the perfect spot for the check. But we don't have an IPFS interface to retrieve the subgraph manifest and verify the network name.

Would adding IPFS as a dependency just for this check be too burdensome for indexers?
What do you think we should do in this situation, @fordN?

Reply:
I agree with your proposed approach, Tiago.  Let's work on validating external subgraph endpoints as a separate task.